### DISCUSSION TITLE
Guide: Troubleshooting 404 errors for GitHub Pages sites

### DISCUSSION THREAD

**Discussion by queenofcorgis** [community/community#61073] [Other Feature Feedback, Questions, & Ideas]
Getting the dreaded 404 error while building a GitHub Page is no fun 👎 and can often be confusing as you work to set up your site. We’ve compiled common reasons you may be seeing the 404 error while working with GitHub Pages. This guide will help you troubleshoot common reasons you may be seeing this 404 error and also lives in our [docs](https://docs.github.com/en/pages/getting-started-with-github-pages/troubleshooting-404-errors-for-github-pages-sites)!   

### GitHub's Status page

If you see a 404 error while building a GitHub Pages site, first check GitHub's [Status page](https://githubstatus.com) for any active incidents. 

### DNS setup

Make sure GitHub's DNS records are set up correctly with your DNS provider. For more information, see [Managing a custom domain for your Pages site](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site) or search in the community. 

### Browser cache

If your Pages site is private and you see a 404 error, you may need to clear your browser's cache. For more information on clearing your cache, see your browser's documentation.

### `index.html` file

GitHub will look for an `index.html` file as the entry file for your site.

* Make sure you have an `index.html` file in the repository for your site on GitHub. For more information, see Creating a GitHub Pages site](https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site#creating-your-site).
* The entry file must be at the top level of your chosen publishing source. For example, if your publishing source is the `/docs` directory on the `main` branch, your entry file must be located in the `/docs` directory on a branch called `main`. 

  If your publishing source is a branch and directory, the entry file must be at the top level of the source directory on the source branch. For example, if your publishing source is the `/docs` directory on the `main` branch, your entry file must be located in the `/docs` directory on a branch called `main`.

  If your publishing source is a GitHub Actions workflow, the artifact that you deploy must include the entry file at the top level of the artifact. Instead of adding the entry file to your repository, you may choose to have your GitHub Actions workflow generate your entry file when the workflow runs.

* The name of the `index.html` file is case sensitive. For example, `Index.html` will not work.
* The name of the file should be `index.html`, not `index.HTML` or any other variation.

### Directory contents

Check that your directory contents are in the root directory.

### Custom domain

If you're using a custom domain, make sure it's set up correctly. For more information, see [About custom domains and GitHub Pages](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/about-custom-domains-and-github-pages). 

* The `CNAME` record should always point to `<USER>.github.io` or `<ORGANIZATION>.github.io`, excluding the repository name.
* If you are able to access your landing page, but encounter broken links throughout, it is likely because you either didn't have a custom domain name before or are reverting back from having a custom domain name. In such cases, changing the routing path does not initiate a rebuild of the page. The recommended solution is to ensure that your site rebuilds automatically when adding or removing a custom domain name. This may involve configuring a commit author and modifying the custom domain name settings.

### Repository

Check whether your repository meets the following requirements.

* The branch you are using to publish your site must be the `main` or default branch.
* The repository must have a commit pushed to it by someone with admin permissions for the repository, such as the repository owner.
* Switching the repository's visibility from public to private or vice versa will change the URL of your GitHub Pages site, which will result in broken links until the site is rebuilt.

### Pages Deployment Update
TL;DR: try adding a `.nojekyll` file to the root of your repo
Special thanks to: 
- @Nectariferous for [some helpful Jekyll tips](https://github.com/orgs/community/discussions/61073#discussioncomment-9993839).
- @datvm For pointing out the Jekyll `_` issues
- @Tryanks for requesting updated documentation

GitHub Pages now uses GitHub Actions to execute the Jekyll build. When using a branch as the source of your build, GitHub Actions must be enabled in your repository if you want to use the built-in Jekyll workflow. Alternatively, if GitHub Actions is unavailable or disabled, adding a `.nojekyll` file to the root of your source branch will bypass the Jekyll build process and deploy the content directly. For more information on enabling GitHub Actions, see "[Managing GitHub Actions settings for a repository](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository)."

**Translation**: If you are using a front end framework other than Jekyll to create your site, and you are running into deployment errors, create a file named `.nojekyll` and put it in the root directory of your `main` branch; then run your build/deployment again. 

[Antora's documentation for deployment](https://docs.antora.org/antora/latest/publish-to-github-pages/#:~:text=nojekyll%20file%20at%20the%20root,file%20when%20you%20run%20Antora.) is excellent, and it can apply to other front end frameworks as well. 
> By default, GitHub Pages runs all files through another static site generator named Jekyll (even if your repository is not set up to use Jekyll). Since Antora already produces a ready-made site, there’s absolutely no need for this step. But it’s more than just the wasted effort.

> Jekyll has the nasty side effect of removing all files that begin with an underscore (_). Why is this a problem? By default, Antora puts UI files in a folder named _. It also places images inside the folder named _images. When Jekyll comes through, it wipes out these folders. As a result, you get no UI and no images.

> .nojekyll
Fortunately, there’s a way to disable this “feature” of GitHub Pages. The solution is to add a .nojekyll file to the root of the published site (i.e., the output directory configured in your playbook).

> The presence of the .nojekyll file at the root of the gh-pages branch tells GitHub Pages not to run the published files through Jekyll. The result is that your Antora-made site will work as expected.

Now that you have this guide, feel free to use it to assist in solving other members’ posts in the GitHub Community and spread the knowledge wealth 🧠. 

Still receiving a 404 error? [Start a discussion](https://github.com/orgs/community/discussions/categories/pages) and share as many details as you can. 

Are we missing any common blockers or you want to see more guides like this? Let us know in the comments!

**Comment by moonfury-ops** [comment]
Thank you

**Reply by kikumon-dev** [reply]
thx

**Reply by RAVIBANKA27** [reply]
I am facing a problem only on uploading index.js file, it shows me 404 error for only index.js file in the inspect window

**Reply by RosmerySI** [reply]
Hi could you find a solution ?I am facing the same problem with js file and also css file

**Comment by Atul9569** [comment]
When you encounter a 404 error on your GitHub Pages site, it usually means that the server couldn't find the requested page. This can happen for a few reasons, but don't worry, we'll work through them step by step.

Here's a guide to help you troubleshoot those pesky 404 errors:

Check Your File Paths: The first thing to do is make sure your file paths are correct. Double-check that you've specified the right file names and folder locations in your URLs and links. Remember, GitHub Pages is case-sensitive, so "Page.html" and "page.html" are considered different files.

Branch and Directory Settings: If your site isn't showing up, make sure that the branch you're using for GitHub Pages is correct. Also, confirm that your site's main HTML file (like "index.html") is in the root directory of your repository.

Wait for Changes to Reflect: If you've recently made changes to your site, remember that it might take a little time for those changes to be visible due to caching. Give it a bit of time, or try clearing your browser cache.

GitHub Pages Build: Sometimes, a 404 error might be caused by a problem during the GitHub Pages build process. Make sure your repository has been successfully built. You can check the "Settings" tab in your repository for GitHub Pages build logs.

Link Errors: If you're linking to external resources like CSS, JavaScript, or images, ensure that the URLs are correct. An incorrect link could lead to a 404 error.

File Extensions: GitHub Pages often assumes that files like "Page.html" should be accessed as "page.html". Check if your file extensions and URLs are consistent.

Repository Permissions: Ensure that your repository is public if you intend for your GitHub Pages site to be accessible to everyone. Private repositories won't be visible to the public.

Custom Domain: If you're using a custom domain, make sure your DNS settings are properly configured. Incorrect settings could lead to 404 errors.

Typos in URLs: It's easy to overlook typos in URLs. Make sure there are no typos in your links and URLs that could be causing the issue.

GitHub Status: Occasionally, GitHub itself might experience issues that can result in temporary disruptions. You can check the GitHub Status page to see if there are any ongoing problems.

By following these steps, you should be able to narrow down the cause of the 404 error on your GitHub Pages site and get it back up and running smoothly.

**Reply by queenofcorgis** [reply]
While we encourage non-AI answers to questions in community, we also love experimenting and using new technologies. We ask if you do use ChatGPT or similar programs, please disclose what software and model version number you used to generate the response you used in your reply in order to promote transparency in the community.

ChatGPT answers in a tone that sounds like a tech support professional and while we support trying out this software, it is important to verify that they are solutions before posting otherwise it may mislead users into thinking that they are receiving official responses when they are not.

Per ChatGPT's website, the program does have limitations like:
- "May occasionally generate incorrect information"
- "May occasionally produce harmful instructions"
- "Limited knowledge of world and events after 2021"

**Comment by datvm** [comment]
Please add another reason is that your folders may start with `_` (underscore), for example Blazor's `_framework` and `_content`. You have to add a `.nojekyll` file into your root directory.

Credit to https://github.com/mpetrovich/stylemark/issues/65.

**Reply by queenofcorgis** [reply]
@datvm thank you for suggesting this addition! This is a new one I haven't heard of so if you start a folder with `_` it must have a `.nojekyll` file in the root directory?

**Reply by datvm** [reply]
@queenofcorgis  Yes. I am not sure why but apparently Github processes the pages through Jekyll which does not output folders that starts with `_`.

With that file in the root directory it doesn't get processed and just output all the files/folders I think. It was kinda a frustrated experiement because all my other files (like `/css/app.css`) work fine but all files in `_framework` folder results in 404.

UPDATE: yep it's confirmed from the [blog post](https://github.blog/2009-12-29-bypassing-jekyll-on-github-pages/):

> It is now possible to completely bypass Jekyll processing on [GitHub Pages](http://pages.github.com/) by creating a file named .nojekyll in the root of your pages repo and pushing it to GitHub. This should only be necessary if your site uses files or directories that start with underscores since Jekyll considers these to be special resources and does not copy them to the final site.

**Comment by AdrianoRean** [comment]
I would suggest to add that if you are using Jekyll, you must know that by default files and directories that start with "_" will be ignored.

**Comment by cpi28** [comment]
I'm trying to deploy a web page with react and vite, I've followed all the steps you mentioned before but I still have a 404 response :/

**Reply by queenofcorgis** [reply]
@cpi28 oh no sorry to hear you're still having trouble! I would suggest opening a new post in this category detailing what you're experiencing and what you've tried so others can assist 👍🏼

**Comment by joominho1234** [comment]
I

**Comment by blackholeapp** [comment]
github.com/user
I need different github.io page with different username

**Comment by anoma3** [comment]
Hello everyone
I hope my messge finds you well.
I am new in github. I have been trying to publish my assignment for a few hours now. The adress I want to visualise should be this one:
https://anoma3.github.io/module5-solution/index.html
It worked for my previous assignments, but this time I get a 404 error and I do not understand why.

My repository is there:
https://github.com/anoma3/module5-solution

Could you help me please?

**Reply by queenofcorgis** [reply]
@anoma3 thanks for posting in the community! I would suggest opening a new post in this category detailing what you're experiencing and what you've tried so others can assist 👍🏼 you're likely to get better visibility that way.

**Comment by Jamancode** [comment]
It's not working! It's driving me crazy!
Ur static page stuff is broken.
static.yml

*Simple workflow for deploying static content to GitHub Pages
name: Deploy static content to Pages

on:
  * Runs on pushes targeting the default branch
  push:
    branches: ["main"]

  * Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

* Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
permissions:
  contents: read
  pages: write
  id-token: write

* Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
* However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
concurrency:
  group: "pages"
  cancel-in-progress: false

jobs:
  * Single deploy job since we're just deploying
  deploy:
    environment:
      name: github-pages
      url: ${{ steps.deployment.outputs.page_url }}
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v4
      - name: Setup Pages
        uses: actions/configure-pages@v4
      - name: Upload artifact
        uses: actions/upload-pages-artifact@v3
        with:
          * Upload entire repository
          path: '.'
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v4
        

I don't know, but it says I don't have the rights to change some things...

Error: .github#L1
actions/checkout@v4, actions/configure-pages@v4, actions/upload-pages-artifact@v3, and actions/deploy-pages@v4 are not allowed to be used in gitmonkey/gitmoneky.github.io. Actions in this workflow must be: within a repository owned by gitmonkey.

IM THE ... OWNER!!! What's happen? 
I've configured all the rights and options, it's on the main branch, and I've checked everything, but none of it seems to be working. It's really frustrating.

Now, I'm utilizing the old version I had stored. However, when attempting to operate with version v4, it feels like colliding my head against the hard wall of reality, as it simply doesn't function.

use this! 
deploy.yml

name: Deploy to GitHub Pages

on:
  push:
    branches:
      - master

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v2
      - name: Install dependencies
        run: |
          npm install
      - name: Build and deploy
        run: |
          npm run build
          npx gh-pages -d build


Please fix this fast! it's the worst for newbies :(

**Reply by LiteBrite82** [reply]
Hi @Jamancode 

I know this is a few months late, but I found an answer [Actions in this workflow must be: within a repository owned by #26609](https://github.com/orgs/community/discussions/26609). 

TL;DR: You need to adjust your account `Actions Permissions` [Managing GitHub Actions permissions for your repository](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#managing-github-actions-permissions-for-your-repository)

![Screenshot 2024-07-17 at 5 10 03 PM](https://github.com/user-attachments/assets/2232335f-c204-482b-9e3e-3774f9961fa1)

**Comment by ECGonzales** [comment]
I am getting a 404 error about my page not being found. It has been working for quite a while and I haven't made any updates to my page since 5 months ago and it all of a sudden stopped working. Has something change in the last few months with the structure?

**Reply by LiteBrite82** [reply]
Hi @ECGonzales 

Sorry I missed your question. 

Did you get any help with this yet? if not, can you please create a new discussion in the Pages category so its easier to track.

**Comment by Tryanks** [comment]
Hi, can you add a guide about `.nojekyll` to the 404 docs and add it with the first post of this discussion? I was trying to deploy my Avalonia WASM project and was held up for a long time until I saw the reply from @datvm which solved the issue. I had also gone through the docs before, but there were no relevant instructions in there :(
This entry would help people like me who are looking to deploy Avalonia/Blazor.

**Reply by LiteBrite82** [reply]
@Tryanks 
Can Do!

**Comment by peddireddyShivareddy** [comment]
I am getting 404 error still everything is made correct, i have index.html and main branch as default still i am getting this error it is worked before. but it is not working now?

**Reply by LiteBrite82** [reply]
Hi @peddireddyShivareddy 

Can you create a new discussion in `pages' for this issue and provide a link to the problem repo?

**Comment by javydevx** [comment]
This guide offers a solid breakdown of common causes behind the 404 error on GitHub Pages, from DNS setup to ensuring the proper placement of the index.html file. The recommendation to add a .nojekyll file is especially helpful for users not relying on Jekyll, as it prevents unwanted file deletions. It's a comprehensive resource for troubleshooting GitHub Pages, but if issues persist, starting a discussion with detailed info is a good next step.

**Comment by JorenVsyntra** [comment]
Hey everyone!

**Comment by dsvcyber** [comment]
I you have an account that is marked as "spam" account you will not be able to publish to Pages and will get that error. I found this out when I set up an account to use for learning and used a passmail generated email. Apparently that account got marked as spam (I did not know this until I attempted to link Git Bash to that GitHub account and there was a banner that popped when authenticated via web browser that stated my account is marked as spam and I'm not able to set up third-party authorizations.
I went back and used an old GitHub account that I had and created a new repository with that site in that account and it worked immediately. I'm not really sure if there's an easier way to tell if your account is marked as a spam account as I don't really see any differences in the settings. I didn't bother raising a ticket as some of the comments in the help pages indicated people have been waiting for weeks to have their accounts removed from being marked as "spam" with no reply from support.

**Comment by AunAby** [comment]
> Getting the dreaded 404 error while building a GitHub Page is no fun 👎 and can often be confusing as you work to set up your site. We’ve compiled common reasons you may be seeing the 404 error while working with GitHub Pages. This guide will help you troubleshoot common reasons you may be seeing this 404 error and also lives in our [docs](https://docs.github.com/en/pages/getting-started-with-github-pages/troubleshooting-404-errors-for-github-pages-sites)!
> 
> ## Troubleshooting 404 errors
> ### GitHub's Status page
> If you see a 404 error while building a GitHub Pages site, first check GitHub's [Status page](https://githubstatus.com) for any active incidents.
> 
> ### DNS setup
> Make sure GitHub's DNS records are set up correctly with your DNS provider. For more information, see [Managing a custom domain for your Pages site](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site) or search in the community.
> 
> ### Browser cache
> If your Pages site is private and you see a 404 error, you may need to clear your browser's cache. For more information on clearing your cache, see your browser's documentation.
> 
> ### `index.html` file
> GitHub will look for an `index.html` file as the entry file for your site.
> 
> * Make sure you have an `index.html` file in the repository for your site on GitHub. For more information, see Creating a GitHub Pages site](https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site#creating-your-site).
> * The entry file must be at the top level of your chosen publishing source. For example, if your publishing source is the `/docs` directory on the `main` branch, your entry file must be located in the `/docs` directory on a branch called `main`.
>   If your publishing source is a branch and directory, the entry file must be at the top level of the source directory on the source branch. For example, if your publishing source is the `/docs` directory on the `main` branch, your entry file must be located in the `/docs` directory on a branch called `main`.
>   If your publishing source is a GitHub Actions workflow, the artifact that you deploy must include the entry file at the top level of the artifact. Instead of adding the entry file to your repository, you may choose to have your GitHub Actions workflow generate your entry file when the workflow runs.
> * The name of the `index.html` file is case sensitive. For example, `Index.html` will not work.
> * The name of the file should be `index.html`, not `index.HTML` or any other variation.
> 
> ### Directory contents
> Check that your directory contents are in the root directory.
> 
> ### Custom domain
> If you're using a custom domain, make sure it's set up correctly. For more information, see [About custom domains and GitHub Pages](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/about-custom-domains-and-github-pages).
> 
> * The `CNAME` record should always point to `<USER>.github.io` or `<ORGANIZATION>.github.io`, excluding the repository name.
> * If you are able to access your landing page, but encounter broken links throughout, it is likely because you either didn't have a custom domain name before or are reverting back from having a custom domain name. In such cases, changing the routing path does not initiate a rebuild of the page. The recommended solution is to ensure that your site rebuilds automatically when adding or removing a custom domain name. This may involve configuring a commit author and modifying the custom domain name settings.
> 
> ### Repository
> Check whether your repository meets the following requirements.
> 
> * The branch you are using to publish your site must be the `main` or default branch.
> * The repository must have a commit pushed to it by someone with admin permissions for the repository, such as the repository owner.
> * Switching the repository's visibility from public to private or vice versa will change the URL of your GitHub Pages site, which will result in broken links until the site is rebuilt.
> 
> ### Pages Deployment Update
> TL;DR: try adding a `.nojekyll` file to the root of your repo Special thanks to:
> 
> * @nectariferous for [some helpful Jekyll tips](https://github.com/orgs/community/discussions/61073#discussioncomment-9993839).
> * @datvm For pointing out the Jekyll `_` issues
> * @Tryanks for requesting updated documentation
> 
> GitHub Pages now uses GitHub Actions to execute the Jekyll build. When using a branch as the source of your build, GitHub Actions must be enabled in your repository if you want to use the built-in Jekyll workflow. Alternatively, if GitHub Actions is unavailable or disabled, adding a `.nojekyll` file to the root of your source branch will bypass the Jekyll build process and deploy the content directly. For more information on enabling GitHub Actions, see "[Managing GitHub Actions settings for a repository](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository)."
> 
> **Translation**: If you are using a front end framework other than Jekyll to create your site, and you are running into deployment errors, create a file named `.nojekyll` and put it in the root directory of your `main` branch; then run your build/deployment again.
> 
> [Antora's documentation for deployment](https://docs.antora.org/antora/latest/publish-to-github-pages/#:~:text=nojekyll%20file%20at%20the%20root,file%20when%20you%20run%20Antora.) is excellent, and it can apply to other front end frameworks as well.
> 
> > By default, GitHub Pages runs all files through another static site generator named Jekyll (even if your repository is not set up to use Jekyll). Since Antora already produces a ready-made site, there’s absolutely no need for this step. But it’s more than just the wasted effort.
> 
> > Jekyll has the nasty side effect of removing all files that begin with an underscore (_). Why is this a problem? By default, Antora puts UI files in a folder named _. It also places images inside the folder named _images. When Jekyll comes through, it wipes out these folders. As a result, you get no UI and no images.
> 
> > .nojekyll
> > Fortunately, there’s a way to disable this “feature” of GitHub Pages. The solution is to add a .nojekyll file to the root of the published site (i.e., the output directory configured in your playbook).
> 
> > The presence of the .nojekyll file at the root of the gh-pages branch tells GitHub Pages not to run the published files through Jekyll. The result is that your Antora-made site will work as expected.
> 
> Now that you have this guide, feel free to use it to assist in solving other members’ posts in the GitHub Community and spread the knowledge wealth 🧠.
> 
> Still receiving a 404 error? [Start a discussion](https://github.com/orgs/community/discussions/categories/pages) and share as many details as you can.
> 
> Are we missing any common blockers or you want to see more guides like this? Let us know in the comments!

**Comment by gopine** [comment]
[Here](https://github.com/rafgraph/spa-github-pages) is a workaround to resolve 404 error for Single Page Applications.
It's just to add the `404.html` and adding a code snippet to the `index.html`.
I've used this in my [repository](https://github.com/gopine/diary-browser) and works perfectly.

**Comment by hroptatyr** [comment]
> If you are able ... In such cases, changing the routing path does not initiate a rebuild of the page. The recommended solution is to ensure that your site rebuilds automatically when adding or removing a custom domain name. This may involve configuring a commit author and modifying the custom domain name settings.

What does that even mean?  I've never "changed the routing path" nor do I know how to "configure a commit author", and it'd be nice if it told me what to modify exactly in the custom domain name settings.

**Comment by muneebhybrid-gif** [comment]
Great article! For high-quality wiring accessories like cable ties, clips, and connectors, you can check out Difvan’s products at [www.difvan.com](http://www.difvan.com/).

**Comment by jsoref** [comment]
@queenofcorgis there's a missing `[` in this article:

<img width="1080" height="634" alt="Screenshot_20251109-230616" src="https://github.com/user-attachments/assets/457b6526-ff24-4fe8-b1a9-711c0311225e" />

**Comment by tade56777** [comment]
Make your web be INDEX.HTML

**Comment by aidasofialily-cmd** [comment]
Make you web with Index.Html

**Comment by wyattverdrell173-blip666** [comment]
> Getting the dreaded 404 error while building a GitHub Page is no fun 👎 and can often be confusing as you work to set up your site. We’ve compiled common reasons you may be seeing the 404 error while working with GitHub Pages. This guide will help you troubleshoot common reasons you may be seeing this 404 error and also lives in our [docs](https://docs.github.com/en/pages/getting-started-with-github-pages/troubleshooting-404-errors-for-github-pages-sites)!
> 
> ## Troubleshooting 404 errors
> ### GitHub's Status page
> If you see a 404 error while building a GitHub Pages site, first check GitHub's [Status page](https://githubstatus.com) for any active incidents.
> 
> ### DNS setup
> Make sure GitHub's DNS records are set up correctly with your DNS provider. For more information, see [Managing a custom domain for your Pages site](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site) or search in the community.
> 
> ### Browser cache
> If your Pages site is private and you see a 404 error, you may need to clear your browser's cache. For more information on clearing your cache, see your browser's documentation.
> 
> ### `index.html` file
> GitHub will look for an `index.html` file as the entry file for your site.
> 
> * Make sure you have an `index.html` file in the repository for your site on GitHub. For more information, see Creating a GitHub Pages site](https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site#creating-your-site).
> * The entry file must be at the top level of your chosen publishing source. For example, if your publishing source is the `/docs` directory on the `main` branch, your entry file must be located in the `/docs` directory on a branch called `main`.
>   If your publishing source is a branch and directory, the entry file must be at the top level of the source directory on the source branch. For example, if your publishing source is the `/docs` directory on the `main` branch, your entry file must be located in the `/docs` directory on a branch called `main`.
>   If your publishing source is a GitHub Actions workflow, the artifact that you deploy must include the entry file at the top level of the artifact. Instead of adding the entry file to your repository, you may choose to have your GitHub Actions workflow generate your entry file when the workflow runs.
> * The name of the `index.html` file is case sensitive. For example, `Index.html` will not work.
> * The name of the file should be `index.html`, not `index.HTML` or any other variation.
> 
> ### Directory contents
> Check that your directory contents are in the root directory.
> 
> ### Custom domain
> If you're using a custom domain, make sure it's set up correctly. For more information, see [About custom domains and GitHub Pages](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/about-custom-domains-and-github-pages).
> 
> * The `CNAME` record should always point to `<USER>.github.io` or `<ORGANIZATION>.github.io`, excluding the repository name.
> * If you are able to access your landing page, but encounter broken links throughout, it is likely because you either didn't have a custom domain name before or are reverting back from having a custom domain name. In such cases, changing the routing path does not initiate a rebuild of the page. The recommended solution is to ensure that your site rebuilds automatically when adding or removing a custom domain name. This may involve configuring a commit author and modifying the custom domain name settings.
> 
> ### Repository
> Check whether your repository meets the following requirements.
> 
> * The branch you are using to publish your site must be the `main` or default branch.
> * The repository must have a commit pushed to it by someone with admin permissions for the repository, such as the repository owner.
> * Switching the repository's visibility from public to private or vice versa will change the URL of your GitHub Pages site, which will result in broken links until the site is rebuilt.
> 
> ### Pages Deployment Update
> TL;DR: try adding a `.nojekyll` file to the root of your repo Special thanks to:
> 
> * @nectariferous for [some helpful Jekyll tips](https://github.com/orgs/community/discussions/61073#discussioncomment-9993839).
> * @datvm For pointing out the Jekyll `_` issues
> * @Tryanks for requesting updated documentation
> 
> GitHub Pages now uses GitHub Actions to execute the Jekyll build. When using a branch as the source of your build, GitHub Actions must be enabled in your repository if you want to use the built-in Jekyll workflow. Alternatively, if GitHub Actions is unavailable or disabled, adding a `.nojekyll` file to the root of your source branch will bypass the Jekyll build process and deploy the content directly. For more information on enabling GitHub Actions, see "[Managing GitHub Actions settings for a repository](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository)."
> 
> **Translation**: If you are using a front end framework other than Jekyll to create your site, and you are running into deployment errors, create a file named `.nojekyll` and put it in the root directory of your `main` branch; then run your build/deployment again.
> 
> [Antora's documentation for deployment](https://docs.antora.org/antora/latest/publish-to-github-pages/#:~:text=nojekyll%20file%20at%20the%20root,file%20when%20you%20run%20Antora.) is excellent, and it can apply to other front end frameworks as well.
> 
> > By default, GitHub Pages runs all files through another static site generator named Jekyll (even if your repository is not set up to use Jekyll). Since Antora already produces a ready-made site, there’s absolutely no need for this step. But it’s more than just the wasted effort.
> 
> > Jekyll has the nasty side effect of removing all files that begin with an underscore (_). Why is this a problem? By default, Antora puts UI files in a folder named _. It also places images inside the folder named _images. When Jekyll comes through, it wipes out these folders. As a result, you get no UI and no images.
> 
> > .nojekyll
> > Fortunately, there’s a way to disable this “feature” of GitHub Pages. The solution is to add a .nojekyll file to the root of the published site (i.e., the output directory configured in your playbook).
> 
> > The presence of the .nojekyll file at the root of the gh-pages branch tells GitHub Pages not to run the published files through Jekyll. The result is that your Antora-made site will work as expected.
> 
> Now that you have this guide, feel free to use it to assist in solving other members’ posts in the GitHub Community and spread the knowledge wealth 🧠.
> 
> Still receiving a 404 error? [Start a discussion](https://github.com/orgs/community/discussions/categories/pages) and share as many details as you can.
> 
> Are we missing any common blockers or you want to see more guides like this? Let us know in the comments!value virtual payments on stripes reverted
